packages/icontools/package.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+# ----------------------------------------------------------------------------
+#     spack install icontools
+#
+# You can edit this file again by typing:
+#
+#     spack edit icontools
+#
+# See the Spack documentation for more information on packaging.
+# ----------------------------------------------------------------------------
+
+from spack import *
+import subprocess
+
+class Icontools(AutotoolsPackage):
+    """
+    DWD ICON Tools for C2SM members. 
+    Set of tools to prepare the input files 
+    (for example the boundary condition, initial condition file,...) for ICON.
+    """
+
+    homepage= 'https://wiki.c2sm.ethz.ch/MODELS/ICONDwdIconTools'
+    git = 'git@github.com:C2SM-ICON/dwd_icon_tools.git'
+
+    maintainers = ['jonasjucker']
+
+    version('master', branch='master')
+
+    depends_on('autoconf', type='build')
+    depends_on('automake', type='build')
+    depends_on('libtool',  type='build')
+    depends_on('m4', type='build')
+
+    depends_on('cray-libsci %cce', type=('build', 'link'))
+    depends_on('netcdf-fortran %cce ~mpi', type=('build', 'link'))
+    depends_on('netcdf-c %cce ~mpi', type=('build', 'link'))
+    depends_on('mpi', type=('build', 'link', 'run'),)
+    depends_on('eccodes ~aec', type=('build', 'link', 'run'))
+    depends_on('cosmo-grib-api', type=('build','link','run'), when='~eccodes')
+    depends_on('hdf5 ~mpi +hl', type=('build','link'))
+    depends_on('jasper@1.900.1%gcc ~shared', type=('build','link'))
+
+    variant('eccodes', default=True, description='Build with eccodes instead of grib-api')
+
+    def configure_args(self):
+        args =[]
+        args.append('CC=cc')
+        args.append('CXX=CC')
```

Comment:
(I am absolutely not sure about this!)...Are you sure this is necessary and not already done by Spack in the build-env?
If not rather do so that it stays as general as possible:
```suggestion
        args.append('CXX=' + self.compiler.cxx)
        etc..
```

Reply:
To pass the compiler as configure arguments was a suggestion from the CSCS-spack person.
So I would keep it.

I agree, your more general suggestion is better. I will implement it that way

Reply:
Using this general solution does not work on Tsa due to linking errors for leccodes_f90.

I removed this section -> Spack should take care of it anyway.